src/main/java/apoc/sequence/Sequence.java

```diff
@@ -0,0 +1,151 @@
+package apoc.sequence;
+
+import apoc.result.NodeResult;
+import apoc.result.ObjectResult;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.procedure.*;
+
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+/**
+ * Created by Valentino Maiorca on 6/29/17.
+ */
+public class Sequence {
+
+    /**
+     * Label for all sequence-node.
+     */
+    public static final Label SEQUENCE_LABEL = Label.label("Apoc_Sequence");
+
+    public static final String PROVIDER_PROPERTY = "provider";
+
+    public static final String NAME_PROPERTY = "name";
+
+    // Simple provider. Increments by 1 the value.
+    public static final SequenceProvider<Long> LONG_SP = new SequenceProvider<Long>() {
+
+        @Override
+        public Long start() {
+            return 0L;
+        }
+
+        @Override
+        public Long next(Long current) {
+            return current + 1;
+        }
+
+        @Override
+        public LinkedList<Long> next(Long current, long length) {
+            return LongStream.range(current + 1, current + length + 1).boxed()
+                    .collect(Collectors.toCollection(LinkedList::new));
+        }
+
+        @Override
+        public String getName() {
+            return "LongCounter";
+        }
+
+    };
+
+    // Map <providerName, sequenceProvider> holding instances of provider for fast access.
+
+    private static final Map<String, SequenceProvider> SEQUENCE_PROVIDER_MAP = new ConcurrentHashMap<>();
+
+    static {
+        SEQUENCE_PROVIDER_MAP.put(LONG_SP.getName(), LONG_SP);
+    }
+
+    /**
+     * Registers a new provider. It will be available for sequence creation.
+     *
+     * @param provider
+     * @return True if the provider wasn't already registered, false otherwise.
+     */
+    public static boolean registerProvider(SequenceProvider provider) {
+        return SEQUENCE_PROVIDER_MAP.put(provider.getName(), provider) == null;
+    }
+
+    @Context
+    public GraphDatabaseService db;
+
+    @UserFunction(value = "apoc.sequence.get")
+    @Description("apoc.sequence.get('sequenceName') - returns the node representing that sequence, null if not existing.")
+    public Node get(@Name("sequenceName") String sequenceName) {
```

Comment:
1. I would never expose this, as it should always  be internal, otherwise uses would be able to mess up
2. I would not use nodes for this for the seame reason but graph properties, either a single "apoc.sequence" graph property with a array content, or better `apoc.sequence.<name>` graph properties with long value